docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,418 @@
+            .unwrap();
+        Ok(Self { node, _publisher, })
+    }
+    fn publish_data(&self,inkrement:i32) -> Result<i32,RclrsError> {
+
+        let msg: StringMsg = StringMsg {
+            data: format!("Hello World {}",inkrement),
+        };
+        self._publisher.publish(msg).unwrap();
+        Ok(inkrement+1_i32)
+    }
+}
+```
+
+1. Implementation Block:
+`impl SimplePublisherNode { ... }`: This line indicates that methods are being defined for the `SimplePublisherNode` struct.
+
+2. Constructor Method:
+* `fn new(context: &Context) -> Result<Self, RclrsError> { ... }`: This method serves as a constructor for creating instances of SimplePublisherNode.
+    * It takes a Context object as input, which is necessary for interacting with the ROS 2 system.
+    * It returns a Result type, indicating either a successful Self (the created SimplePublisherNode object) or an RclrsError if something goes wrong.
+    * Inside the new method:
+        * `let node = create_node(context, "simple_publisher").unwrap();`: Creates a new ROS 2 node named "simple_publisher" within the given context. The unwrap() unwraps the result, handling any errors immediately.
+        * `let _publisher = node.create_publisher("publish_hello", QOS_PROFILE_DEFAULT).unwrap();`: Creates a publisher for string messages on the topic "publish_hello" with default quality of service settings.
+        * `Ok(Self { node, _publisher, })`: Returns a Result with the newly created SimplePublisherNode object, containing the node and publisher references.
+
+3. Publishing Method:
+* `fn publish_data(&self, inkrement: i32) -> Result<i32, RclrsError> { ... }`: This method publishes a string message and increments a counter.
+    * It takes an inkrement value (an integer) as input, which is likely used for counting purposes within the message content.
+    * It also returns a Result type, indicating either the incremented inkrement value or an RclrsError if publishing fails.
+    * Inside the publish_data method:
+        * `let msg: StringMsg = StringMsg { data: format!("Hello World {}", inkrement), };`:tCreates a string message with the content "Hello World" followed by the inkrement value.
+        * self._publisher.publish(msg).unwrap();: Publishes the created message onto the topic associated with the publisher.
+        * Ok(inkrement + 1_i32): Returns a Result with the incremented inkrement value.  
+
+#### The main Method creates a ROS 2 node that publishes string messages at a rate of 1 Hz.
+
+```
+fn main() -> Result<(),RclrsError> {
+   let context = Context::new(std::env::args()).unwrap();
+   let publisher = Arc::new(SimplePublisherNode::new(&context).unwrap());
+   let publisher_other_thread = Arc::clone(&publisher);
+   let mut iterator: i32=0;
+   thread::spawn(move || -> () {
+       iter::repeat(()).for_each(|()| {
+           thread::sleep(Duration::from_millis(1000));
+           iterator=publisher_other_thread.publish_data(iterator).unwrap();
```

Comment:
Again, I feel like "count" or "counter" might be a better variable name than "iterator" here. Especially because we're already using an iterator on line 248.

Reply:
done